types/Config.js

```diff
@@ -181,3 +183,74 @@ export type ProjectConfig = {|
   transformIgnorePatterns: Array<Glob>,
   unmockedModulePathPatterns: ?Array<string>,
 |};
+
+export type Argv = {
```

Comment:
Do you mind reordering this file in terms of how it is flowing through Jest? Something like this:

Argv
DefaultOptions
InitialOptions
GlobalConfig
ProjectConfig

I think that makes it a bit more readable. We could also consider having a separate `types/Options` file for Argv and DefaultOptions/InitialOptions.